spec/parsers/zip_parser_spec.rb

```diff
@@ -103,4 +103,12 @@
     expect(first_entry.filename).to eq('Li��nia Extreme//')
     expect(first_entry.type).to eq(:directory)
   end
+
+  it 'is able to handle files with invalid central directory position' do
+    invalid_zip_path = fixtures_dir + '/ZIP/invalid_central_directory.zip'
+
+    result = subject.call(File.open(invalid_zip_path, 'rb'))
+
+    expect(result).to be_nil
```

Comment:
At first sight, it feels odd that the test for a piece of code that raises an exception returns `nil` 🤔

Reply:
you're right! I changed to:

```ruby
expect { subject.call(File.open(invalid_zip_path, 'rb')) }
  .to_not raise_error
```